test/PoolFactory/deployPool.spec.ts

```diff
@@ -215,4 +217,51 @@ describe("PoolFactory.deployPool", () => {
             )
         })
     })
+    context("Clone deterministic checks", async () => {
+        let cloneLibrary: TestClones
+
+        before(async () => {
+            signers = await ethers.getSigners()
+            const cloneLibraryFactory = (await ethers.getContractFactory(
+                "TestClones",
+                signers[0]
+            )) as TestClones__factory
+            cloneLibrary = await cloneLibraryFactory.deploy()
+            await cloneLibrary.deployed()
+        })
+
+        it("should deploy deterministically", async () => {
+            let encoder = new ethers.utils.AbiCoder()
+            let abiEncoded = encoder.encode(
+                ["uint16", "address", "address"],
+                [leverage, token.address, oracleWrapper.address]
+            )
+            let uniqueIdHash = ethers.utils.keccak256(abiEncoded)
+            let predictedPoolAddress =
+                await cloneLibrary.predictDeterministicAddress(
+                    await factory.poolBaseAddress(),
+                    uniqueIdHash,
+                    factory.address
+                )
+
+            expect(predictedPoolAddress).to.eq(pool.address)
+        })
+
+        it("should not equal of leverage is different", async () => {
```

Comment:
Unsure, but is this the intent?
```suggestion
        it("should not equal if leverage is different", async () => {
```

Reply:
yea